File: Dopamine.CollectionModule/CollectionModule.vb
Imports Microsoft.Practices.Prism.Modularity
Imports Microsoft.Practices.Prism.Regions
Imports Microsoft.Practices.Unity
Imports Dopamine.CollectionModule.ViewModels
Imports Dopamine.CollectionModule.Views
Imports Dopamine.Core.Prism
Imports Dopamine.ControlsModule.Views
Imports Dopamine.Core.Settings

Public Class CollectionModule
    Implements IModule

#Region "Variables"
    Private ReadOnly mRegionManager As IRegionManager
    Private mContainer As IUnityContainer
#End Region

#Region "Construction"
    Public Sub New(iRegionManager As IRegionManager, iContainer As IUnityContainer)
        Me.mRegionManager = iRegionManager
        Me.mContainer = iContainer
    End Sub
#End Region

#Region "Functions"
    Public Sub Initialize() Implements IModule.Initialize
        Me.mContainer.RegisterType(Of Object, CollectionViewModel)(GetType(CollectionViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, Collection)(GetType(Collection).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionSubMenu)(GetType(CollectionSubMenu).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionAlbumsViewModel)(GetType(CollectionAlbumsViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionAlbums)(GetType(CollectionAlbums).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionArtistsViewModel)(GetType(CollectionArtistsViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionArtists)(GetType(CollectionArtists).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionGenresViewModel)(GetType(CollectionGenresViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionGenres)(GetType(CollectionGenres).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionPlaylistsViewModel)(GetType(CollectionPlaylistsViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionPlaylists)(GetType(CollectionPlaylists).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionTracksViewModel)(GetType(CollectionTracksViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionTracks)(GetType(CollectionTracks).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionCloudViewModel)(GetType(CollectionCloudViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionCloud)(GetType(CollectionCloud).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionTracksColumnsViewModel)(GetType(CollectionTracksColumnsViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, CollectionTracksColumns)(GetType(CollectionTracksColumns).FullName)

        Me.mRegionManager.RegisterViewWithRegion(RegionNames.SubMenuRegion, GetType(CollectionSubMenu))
        Me.mRegionManager.RegisterViewWithRegion(RegionNames.ContentRegion, GetType(Collection))

        Me.mRegionManager.RegisterViewWithRegion(RegionNames.CollectionPlaybackControlsRegion, GetType(CollectionPlaybackControls))
        Me.mRegionManager.RegisterViewWithRegion(RegionNames.CollectionSpectrumAnalyzerRegion, GetType(SpectrumAnalyzerControl))

        If XmlSettingsClient.Instance.Get(Of Boolean)("Startup", "ShowLastSelectedPage") Then
            Dim screen As SelectedPage = CType(XmlSettingsClient.Instance.Get(Of Integer)("FullPlayer", "SelectedPage"), SelectedPage)

            Select Case screen
                Case SelectedPage.Artists
                    Me.mRegionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, GetType(CollectionArtists))
                Case SelectedPage.Genres
                    Me.mRegionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, GetType(CollectionGenres))
                Case SelectedPage.Albums
                    Me.mRegionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, GetType(CollectionAlbums))
                Case SelectedPage.Tracks
                    Me.mRegionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, GetType(CollectionTracks))
                Case SelectedPage.Playlists
                    Me.mRegionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, GetType(CollectionPlaylists))
                Case SelectedPage.Recent
                    Me.mRegionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, GetType(CollectionCloud))
                Case Else
                    Me.mRegionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, GetType(CollectionArtists))
            End Select
        Else
            Me.mRegionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, GetType(CollectionArtists))
        End If

    End Sub
#End Region
End Class

File: Dopamine.CollectionModule/CollectionModule.cs
using Digimezzo.Utilities.Settings;
using Dopamine.CollectionModule.ViewModels;
using Dopamine.CollectionModule.Views;
using Dopamine.Common.Enums;
using Dopamine.ControlsModule.Views;
using Dopamine.Common.Prism;
using Microsoft.Practices.Unity;
using Prism.Modularity;
using Prism.Regions;

namespace Dopamine.CollectionModule
{
    public class CollectionModule : IModule
    {
        #region Variables
        private readonly IRegionManager regionManager;
        private IUnityContainer container;
        #endregion

        #region Construction
        public CollectionModule(IRegionManager regionManager, IUnityContainer container)
        {
            this.regionManager = regionManager;
            this.container = container;
        }
        #endregion

        #region Functions
        public void Initialize()
        {
            this.container.RegisterType<object, CollectionViewModel>(typeof(CollectionViewModel).FullName);
            this.container.RegisterType<object, Collection>(typeof(Collection).FullName);
            this.container.RegisterType<object, CollectionSubMenu>(typeof(CollectionSubMenu).FullName);
            this.container.RegisterType<object, CollectionAlbumsViewModel>(typeof(CollectionAlbumsViewModel).FullName);
            this.container.RegisterType<object, CollectionAlbums>(typeof(CollectionAlbums).FullName);
            this.container.RegisterType<object, CollectionArtistsViewModel>(typeof(CollectionArtistsViewModel).FullName);
            this.container.RegisterType<object, CollectionArtists>(typeof(CollectionArtists).FullName);
            this.container.RegisterType<object, CollectionGenresViewModel>(typeof(CollectionGenresViewModel).FullName);
            this.container.RegisterType<object, CollectionGenres>(typeof(CollectionGenres).FullName);
            this.container.RegisterType<object, CollectionPlaylistsViewModel>(typeof(CollectionPlaylistsViewModel).FullName);
            this.container.RegisterType<object, CollectionPlaylists>(typeof(CollectionPlaylists).FullName);
            this.container.RegisterType<object, CollectionTracksViewModel>(typeof(CollectionTracksViewModel).FullName);
            this.container.RegisterType<object, CollectionTracks>(typeof(CollectionTracks).FullName);
            this.container.RegisterType<object, CollectionCloudViewModel>(typeof(CollectionCloudViewModel).FullName);
            this.container.RegisterType<object, CollectionCloud>(typeof(CollectionCloud).FullName);
            this.container.RegisterType<object, CollectionTracksColumnsViewModel>(typeof(CollectionTracksColumnsViewModel).FullName);
            this.container.RegisterType<object, CollectionTracksColumns>(typeof(CollectionTracksColumns).FullName);

            this.regionManager.RegisterViewWithRegion(RegionNames.SubMenuRegion, typeof(CollectionSubMenu));
            this.regionManager.RegisterViewWithRegion(RegionNames.ContentRegion, typeof(Collection));

            this.regionManager.RegisterViewWithRegion(RegionNames.CollectionPlaybackControlsRegion, typeof(CollectionPlaybackControls));
            this.regionManager.RegisterViewWithRegion(RegionNames.CollectionSpectrumAnalyzerRegion, typeof(SpectrumAnalyzerControl));

            if (SettingsClient.Get<bool>("Startup", "ShowLastSelectedPage"))
            {
                SelectedPage page = (SelectedPage)SettingsClient.Get<int>("FullPlayer", "SelectedPage");

                switch (page)
                {
                    case SelectedPage.Artists:
                        this.regionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, typeof(CollectionArtists));
                        break;
                    case SelectedPage.Genres:
                        this.regionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, typeof(CollectionGenres));
                        break;
                    case SelectedPage.Albums:
                        this.regionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, typeof(CollectionAlbums));
                        break;
                    case SelectedPage.Tracks:
                        this.regionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, typeof(CollectionTracks));
                        break;
                    case SelectedPage.Playlists:
                        this.regionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, typeof(CollectionPlaylists));
                        break;
                    case SelectedPage.Recent:
                        this.regionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, typeof(CollectionCloud));
                        break;
                    default:
                        this.regionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, typeof(CollectionArtists));
                        break;
                }
            }
            else
            {
                this.regionManager.RegisterViewWithRegion(RegionNames.CollectionContentRegion, typeof(CollectionArtists));
            }

        }
        #endregion
    }
}
